File: app/models/recipe.rb
class Recipe < ApplicationRecord
  belongs_to :author
  belongs_to :image, optional: true

  has_many :tagged_recipes
  has_many :tags, through: :tagged_recipes
  has_many :ingredients

  enum :difficulty, { very_easy: 0, easy: 1, medium: 2, hard: 3 }
  enum :budget, { cheap: 0, reasonable: 1, expensive: 2 }
end

File: app/javascript/components/Recipes/types/recipe.ts
import Tag from "./tag";
import Ingredient from "./ingredient";
import Author from "./author";
import Image from "./image";

type Recipe = {
  name: string;
  tip: string;
  difficulty: string;
  budget: string;
  rate: number;
  quantity_of_people: number;
  prep_time: string;
  cook_time: string;
  total_time: string;
  image?: Image;
  author: Author;
  tags: Tag[];
  ingredients: Ingredient[];
};

export default Recipe;
